calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,309 @@
+
+    def _generate_profile_json(self, profile_name, rules):
+        """
+        Given a list of of Calico rules, generate a Calico Profile JSON blob
+        implementing those rules.
+
+        :param profile_name: The name of the Calico profile
+        :type profile_name: string
+        :param rules: A tuple of (inbound, outbound) Calico rules
+        :type rules: tuple
+        :return: A JSON blob ready to be loaded by calicoctl
+        :rtype: str
+        """
+        inbound, outbound = rules
+        profile = {
+            'id': profile_name,
+            'inbound_rules': inbound,
+            'outbound_rules': outbound,
+        }
+        profile_json = json.dumps(profile, indent=2)
+        print('Final profile "%s": %s' % (profile_name, profile_json))
+        return profile_json
+
+    def _apply_rules(self, profile_name, pod):
+        """
+        Generate a new profile with the specified rules.
+
+        This contains inbound allow rules for all the ports we gathered,
+        plus a default 'allow from <profile_name>' to allow traffic within a
+        profile group.
+
+        :param profile_name: The profile to update
+        :type profile_name: string
+        :param pod: The config dictionary for the pod being created.
+        :type pod: dict
+        :return:
+        """
+        rules = self._generate_rules()
+        profile_json = self._generate_profile_json(profile_name, rules)
+
+        # Pipe the Profile JSON into the calicoctl command to update the rule.
+        calicoctl('profile', profile_name, 'rule', 'update', _in=profile_json)
+        print('Finished applying rules.')
+
+    def _apply_tags(self, profile_name, pod):
+        """
+        Extract the label KV pairs from the pod config, and apply each as a
+        tag in the pod's profile.
+
+        :param profile_name: The name of the Calico profile.
+        :type profile_name: string
+        :param pod: The config dictionary for the pod being created.
+        :type pod: dict
+        :return:
+        """
+        try:
+            labels = pod['metadata']['labels']
+        except KeyError:
+            # If there are no labels, there's no more work to do.
+            print('No labels found in pod %s' % pod)
+            return
+
+        for k, v in labels.iteritems():
+            tag = '%s_%s' % (k, v)
+            tag = tag.replace('-', '_')
+            print('Adding tag ' + tag)
+            try:
+                calicoctl('profile', profile_name, 'tag', 'add', tag)
+            except sh.ErrorReturnCode as e:
+                print('Could not create tag %s.\n%s' % (tag, e))
+        print('Finished applying tags.')
+
+if __name__ == '__main__':
+    print('Args: %s' % sys.argv)
+    mode = sys.argv[1]
+
+    if mode == 'init':
+        print('No initialization work to perform')
+    elif mode == 'setup':
+        print('Executing Calico pod-creation hook')
+        NetworkPlugin().create(sys.argv)
```

Comment:
I think this code should be parsing the command line arguments and passing them to the correct methods, rather than just blindly passing the full args list - creates a better separation of integration code and plugin code, and makes the above methods easier to read.